### PR TITLE
ItemPackage no longer crashes when backing content is deleted.

### PR DESCRIPTION
**Pull request by nickmshelley**

### CODE REVIEW

**Comment by nickmshelley** [comment]
Tests pass locally

File: LDSContent/ContentController.swift [hunk 1]
```diff
@@ -247,7 +249,11 @@ extension ContentController {
     /// The currently installed item package for the designated item.
     public func itemPackageForItemWithID(_ itemID: Int64) -> ItemPackage? {
         if let installedVersion = contentInventory.installedVersionOfItemWithID(itemID) {
-            return try? ItemPackage(url: location.appendingPathComponent("Item/\(itemID)/\(installedVersion.schemaVersion).\(installedVersion.itemPackageVersion)"))
+            let package = try? ItemPackage(url: location.appendingPathComponent("Item/\(itemID)/\(installedVersion.schemaVersion).\(installedVersion.itemPackageVersion)"))
```
Comment (stephanheilner) [inline LDSContent/ContentController.swift]:
```swift
if let package = try? ItemPackage(url: location.appendingPathComponent("Item/\(itemID)/\(installedVersion.schemaVersion).\(installedVersion.itemPackageVersion)")) {
    itemPackageDatabaseDeletedObservers.add(package, ItemPackage.handleContentDeleted)
    return package 
}
return nil
```

File: LDSContent/ContentController.swift [hunk 2]
```diff
@@ -43,6 +43,7 @@ public class ContentController {
     public let itemPackageUninstallObservers = ObserverSet<Item>()
     public let itemPackageInstallProgressObservers = ObserverSet<(item: Item, progress: Float)>()
     public let networkActivityObservers = ObserverSet<NetworkActivity>()
+    public let itemPackageDatabaseDeletedObservers = ObserverSet<URL>()
```
Comment (stephanheilner) [inline LDSContent/ContentController.swift]: This could maybe be named: `itemPackageDeletedObservers`

File: LDSContent/ItemPackage.swift [hunk 3]
```diff
@@ -27,28 +27,34 @@ import FTS3HTMLTokenizer
...
             // Only disable foreign keys when using as a readonly database for better performance
-            try db.execute("PRAGMA foreign_keys = OFF")
+            try db?.execute("PRAGMA foreign_keys = OFF")
         }
         
-        registerTokenizer(db.handle, UnsafeMutablePointer<Int8>(mutating: ("HTMLTokenizer" as NSString).utf8String))
+        registerTokenizer(db?.handle, UnsafeMutablePointer<Int8>(mutating: ("HTMLTokenizer" as NSString).utf8String))
+    }
+    
+    func handleContentDeleted(url: URL) {
```
Comment (stephanheilner) [inline LDSContent/ItemPackage.swift]: Maybe name this: `itemPackageDeleted`

**Pull request merged**